File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/presenter/MainPresenter.java
package com.github.batulovandrey.unofficialurbandictionary.presenter;

import com.github.batulovandrey.unofficialurbandictionary.adapter.DefinitionAdapter;
import com.github.batulovandrey.unofficialurbandictionary.adapter.DefinitionClickListener;

/**
 * @author Andrey Batulov on 27/10/2017
 */

public interface MainPresenter {

    void showDataInRecycler();

    void showQueriesInListView();

    boolean textChanged(String text);

    void filterText(String text);

    void setAdapterToRecycler(DefinitionAdapter definitionAdapter);

    void getData(String query, DefinitionClickListener listener);

    void showToast(int resId);

    long getDefinitionId(int position);

    void showProgressbar();

    void hideProgressbar();
}

File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/presenter/MainPresenter.kt
package com.github.batulovandrey.unofficialurbandictionary.presenter

import com.github.batulovandrey.unofficialurbandictionary.adapter.DefinitionAdapter
import com.github.batulovandrey.unofficialurbandictionary.adapter.DefinitionClickListener
import com.github.batulovandrey.unofficialurbandictionary.adapter.QueriesAdapter

/**
 * @author Andrey Batulov on 22/12/2017
 */

interface MainPresenter {

    fun showDefinitionsInRecycler()

    fun showQueriesInRecycler()

    fun textChanged(text: String): Boolean

    fun setAdapterToDefinitionsRecycler(definitionAdapter: DefinitionAdapter)

    fun setAdapterToQueriesRecycler(queriesAdapter: QueriesAdapter)

    fun getData(query: String, listener: DefinitionClickListener)

    fun showToast(resId: Int)

    fun getDefinitionId(position: Int): Long

    fun showProgressbar()

    fun hideProgressbar()

    fun saveQueryToRealm(query: String)

    fun initializeQueryToServer(query: String)
}